### PR TITLE
ExceptionReceivedEventArgs.Action consolidation

### PR DESCRIPTION
**Pull request by SeanFeldman**
Fixes issue #174 

@nemakam or vinaysurya please review

@sethmanheim these are the actions for the **new client**, not sure about the old/current one

### CODE REVIEW

File: src/Microsoft.Azure.ServiceBus/ExceptionReceivedEventArgsAction.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿namespace Microsoft.Azure.ServiceBus
+{
+    static class ExceptionReceivedEventArgsAction
```
Comment (nemakam) [inline src/Microsoft.Azure.ServiceBus/ExceptionReceivedEventArgsAction.cs]: what are your thoughts on making this public?

Comment (SeanFeldman) [inline src/Microsoft.Azure.ServiceBus/ExceptionReceivedEventArgsAction.cs]: I assume you're asking me @nemakam. I'm on a fence about that. What would be the a strong use case for having it public?

Comment (nemakam) [inline src/Microsoft.Azure.ServiceBus/ExceptionReceivedEventArgsAction.cs]: I'm sure it doesn't fit into the mainline scenario. But even for a very simple use case wherein a customer wants to keep a count of exceptions occurred during each operation, he would need to segregate all the exceptions received based on this action.

**Comment by SeanFeldman** [comment]
@nemakam made it public. Might need help with XML doc to describe better those actions.
image

File: src/Microsoft.Azure.ServiceBus/ExceptionReceivedEventArgsAction.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
+﻿namespace Microsoft.Azure.ServiceBus
+{
+    using Microsoft.Azure.ServiceBus.Primitives;
+
+    /// <summary>Action taking place when <see cref="ExceptionReceivedEventArgs"/> is raised for a processed message.</summary>
```
Comment (nemakam) [inline src/Microsoft.Azure.ServiceBus/ExceptionReceivedEventArgsAction.cs]: > is raised for a processed message.

This doesn't sound right. We have actions here like `AcceptMessageSession` and `CloseMessageSession` and `Receive`

Comment (SeanFeldman) [inline src/Microsoft.Azure.ServiceBus/ExceptionReceivedEventArgsAction.cs]: Ha-ha, hence [the comment](https://github.com/Azure/azure-service-bus-dotnet/pull/176#issuecomment-306372568): 

> Might need help with XML doc to describe better those actions.

Comment (SeanFeldman) [inline src/Microsoft.Azure.ServiceBus/ExceptionReceivedEventArgsAction.cs]: btw, I tweaked it a little. Will let the team come up with a better xml summaries.

Comment (nemakam) [inline src/Microsoft.Azure.ServiceBus/ExceptionReceivedEventArgsAction.cs]: Okay, but meanwhile could you please remove the 'processed message' part? Incomplete documentation isn't as bad as wrong documentation :)

**Comment by SeanFeldman** [comment]
@nemakam done

**Pull request merged**